src/main/java/hudson/plugins/sonar/MsBuildSQRunnerInstallation.java

```diff
@@ -0,0 +1,130 @@
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.StaplerRequest;
+import hudson.model.EnvironmentSpecific;
+import hudson.slaves.NodeSpecific;
+import hudson.tools.ToolInstallation;
+import jenkins.security.MasterToSlaveCallable;
+import hudson.Launcher;
+import hudson.Extension;
+import hudson.tools.ToolDescriptor;
+import hudson.tools.ToolInstaller;
+import hudson.Util;
+import hudson.tools.ToolProperty;
+import org.kohsuke.stapler.DataBoundConstructor;
+import hudson.EnvVars;
+import hudson.model.TaskListener;
+import hudson.model.Node;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+public class MsBuildSQRunnerInstallation extends ToolInstallation implements EnvironmentSpecific<MsBuildSQRunnerInstallation>, NodeSpecific<MsBuildSQRunnerInstallation> {
+  private static final String EXE_NAME = "MSBuild.SonarQube.Runner.exe";
+  private static final long serialVersionUID = 1L;
+  private final String exeName;
+
+  @DataBoundConstructor
+  public MsBuildSQRunnerInstallation(String name, String home, List<? extends ToolProperty<?>> properties) {
+    this(Util.fixEmptyAndTrim(name), Util.fixEmptyAndTrim(home), properties, EXE_NAME);
+  }
+
+  @VisibleForTesting
+  public MsBuildSQRunnerInstallation(String name, String home, List<? extends ToolProperty<?>> properties, String exeName) {
+    super(Util.fixEmptyAndTrim(name), Util.fixEmptyAndTrim(home), properties);
+    this.exeName = exeName;
+  }
+
+  @Override
+  public MsBuildSQRunnerInstallation forEnvironment(EnvVars environment) {
+    return new MsBuildSQRunnerInstallation(getName(), environment.expand(getHome()), getProperties().toList());
+  }
+
+  @Override
+  public MsBuildSQRunnerInstallation forNode(Node node, TaskListener log) throws IOException, InterruptedException {
+    return new MsBuildSQRunnerInstallation(getName(), translateFor(node, log), getProperties().toList());
+  }
+
+  @Extension
+  public static class DescriptorImpl extends ToolDescriptor<MsBuildSQRunnerInstallation> {
+    // super already manages: persistence of installations and form json deserialization
+    public DescriptorImpl() {
+      super();
+      load();
+    }
+
+    @Override
+    public void setInstallations(MsBuildSQRunnerInstallation... installations) {
+      super.setInstallations(installations);
+      save();
+    }
+
+    @Override
+    public MsBuildSQRunnerInstallation newInstance(StaplerRequest req, JSONObject formData) {
+      return (MsBuildSQRunnerInstallation) req.bindJSON(clazz, formData);
+    }
+
+    @Override
+    public String getDisplayName() {
+      return "MSBuild SonarQube Runner";
+    }
+
+    @Override
+    public List<? extends ToolInstaller> getDefaultInstallers() {
+      return Collections.singletonList(new MsBuildSonarQubeRunnerInstaller(null));
+    }
+  }
+
+  public String getExecutable(Launcher launcher) throws IOException, InterruptedException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.lang.InterruptedException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1160)